test/e2e/serial/tests/non_regression.go

```diff
@@ -421,4 +422,115 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload placeme
+			neededNodes := 2
+
+			// random "high enough" value to let the infra pods run comfortably
+			expectedFreeRes := corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("6"),
+				corev1.ResourceMemory: resource.MustParse("6Gi"),
+			}
+
+			nrtCandidates := e2enrt.FilterAnyZoneMatchingResources(nrts, expectedFreeRes)
+			if len(nrtCandidates) < neededNodes {
+				e2efixture.Skipf(fxt, "not enough nodes with 2 NUMA Zones: found %d, needed %d", len(nrtCandidates), neededNodes)
+			}
+			// note we assume all NRT zones are equal, then we take the most accessible
+			refZone := nrts[0].Zones[0]
+			// abuse the function to compute the resources for the workload, not for the padding pods
+			// the desired result is to have workload pods which almost saturate a NUMA zone.
+			requiredRes, err := e2enrt.SaturateZoneUntilLeft(refZone, expectedFreeRes, e2enrt.DropHostLevelResources)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("waiting for the NRT data to settle")
+			e2efixture.MustSettleNRT(fxt)
+
+			dpName := "test-dp-scaleup"
+			replicas := int32(len(nrtCandidates)) // initially. Then the final value will be 1 replica per NUMA node
+			By(fmt.Sprintf("creating a deployment %q with replicas %d candidate nodes %d", dpName, replicas, len(nrtCandidates)))
+
+			nroSchedObj := nrosched.CheckNROSchedulerAvailable(ctx, fxt.Client, objectnames.DefaultNUMAResourcesSchedulerCrName)
+
+			podLabels := map[string]string{
+				"test": dpName,
+			}
+
+			schedulerName := nroSchedObj.Status.SchedulerName // shortcut
+			podSpec := corev1.PodSpec{
+				SchedulerName: schedulerName,
+				Containers: []corev1.Container{
+					{
+						Name:  dpName + "-cnt",
+						Image: images.GetPauseImage(),
+						Command: []string{
+							images.PauseCommand,
+						},
+						Resources: corev1.ResourceRequirements{
+							Requests: requiredRes,
+							Limits:   requiredRes,
+						},
+					},
+				},
+				RestartPolicy: corev1.RestartPolicyAlways,
+			}
+			dp := objects.NewTestDeploymentWithPodSpec(replicas, podLabels, nodeSelector, fxt.Namespace.Name, dpName, podSpec)
+			Expect(fxt.Client.Create(ctx, dp)).To(Succeed())
+
+			// although the deployment pods will be pending thus the deployment will not be counted as complete,
+			// we need to wait until all the replicas are created despite their status before moving forward with the checks
+			By("wait for the deployment to be up with its pod created")
+			dp, err = wait.With(fxt.Client).Interval(time.Second).Timeout(time.Minute).ForDeploymentReplicasReadiness(ctx, dp, replicas)
+			Expect(err).NotTo(HaveOccurred())
+
+			By(fmt.Sprintf("checking deployment pods have been scheduled by %q ", schedulerName))
+			pods, err := podlist.With(fxt.Client).ByDeployment(ctx, *dp)
+			Expect(err).ToNot(HaveOccurred(), "unable to get pods from deployment %q:  %v", dp.Name, err)
+			Expect(pods).ToNot(BeEmpty(), "cannot find any pods for DP %s/%s", dp.Namespace, dp.Name)
+
+			var errorPods []string
```

Comment:
or make that a single error but use error.Join to extend it:
https://github.com/openshift-kni/numaresources-operator/blob/main/pkg/validation/validation.go#L56

Reply:
it's treated as counter atm, will refactor as such.